mantidproject/mantid
# SampleLogs filtering

**Description of work.**

Added the possibility of filtering SampleLogs using an added QLineEdit. When editing is finished, the model is recreated and only the keys that match (case-insensitive) the given string are displayed. 

**To test:**

1. In the Workbench, load any data with logs you would like, e.g. ILL/D22/192068.nxs from the UnitTest data
2. Right-click on the created workspace and choose Show Sample Logs
3. Notice there is a new QLineEdit widget at the bottom of the SampleLogs dialog window.
4. Type-in a keyword, e.g. 'beam' for ILL/D22/192068.nxs, press Enter and check whether all the shown logs contain this keyword. Please notice that the matching is case-insensitive.
5. Repeat with a couple of different keywords and check if the shown logs match the typed-in keyword. 

Fixes #30090 . 

*Release notes* added.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
